serializer.rst

```diff
@@ -2,223 +2,443 @@ How to Use the Serializer
+        class PersonController extends AbstractController
+        {
+            // ...
 
-.. _serializer_serializer-context:
+            public function create(Request $request, SerializerInterface $serializer): Response
+            {
+                if ('json' !== $request->getContentTypeFormat()) {
+                    throw new BadRequestException('Unsupported content format');
+                }
 
-Serializer Context
-------------------
+                $jsonData = $request->getContent();
+                $person = $serializer->deserialize($jsonData, Person::class, 'json');
 
-The serializer can define a context to control the (de)serialization of
-resources. This context is passed to all normalizers. For example:
+                // ... do something with $person and return a response
+            }
+        }
 
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\DateTimeNormalizer` uses
-  ``datetime_format`` key as date time format;
-* :class:`Symfony\\Component\\Serializer\\Normalizer\\AbstractObjectNormalizer`
-  uses ``preserve_empty_objects`` to represent empty objects as ``{}`` instead
-  of ``[]`` in JSON.
-* :class:`Symfony\\Component\\Serializer\\Serializer`
-  uses ``empty_array_as_object`` to represent empty arrays as ``{}`` instead
-  of ``[]`` in JSON.
+    .. code-block:: php-standalone
 
-.. versionadded:: 5.4
+        use App\Model\Person;
+        use Symfony\Component\Serializer\Encoder\JsonEncoder;
+        use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+        use Symfony\Component\Serializer\Serializer;
 
-    The usage of the ``empty_array_as_object`` option in the
-    Serializer was introduced in Symfony 5.4.
+        // ...
+        $jsonData = ...; // fetch JSON from the request
+        $person = $serializer->deserialize($jsonData, Person::class, 'json');
 
-You can pass the context as follows::
+In this case, :method:`Symfony\\Component\\Serializer\\Serializer::deserialize`
+needs three parameters:
 
-    $serializer->serialize($something, 'json', [
-        DateTimeNormalizer::FORMAT_KEY => 'Y-m-d H:i:s',
-    ]);
+#. The data to be decoded
+#. The name of the class this information will be decoded to
+#. The name of the encoder used to convert the data to an array (i.e. the
+   input format)
 
-    $serializer->deserialize($someJson, Something::class, 'json', [
-        DateTimeNormalizer::FORMAT_KEY => 'Y-m-d H:i:s',
-    ]);
+When sending a request to this controller (e.g.
+``{"first_name":"John Doe","age":54,"sportsperson":true}``), the serializer
+will create a new instance of ``Person`` and sets the properties to the
+values from the given JSON.
+
+.. note::
+
+    By default, additional attributes that are not mapped to the
+    denormalized object will be ignored by the Serializer component. For
+    instance, if a request to the above controller contains ``{..., "city": "Paris"}``,
+    the ``city`` field will be ignored. You can disallow extra attributes
```

Comment:
What does this mean? --> "You can disallow extra attributes"

"Disallow" means that the app will throw an exception if it finds non-mapped attributes?

Reply:
Correct, I've clarified it a bit.